### PR TITLE
vicuna

### PR DESCRIPTION
**Pull request by aglick24**
added a to_chatml function for vicuna and testing the fork, will next do a from_chatml function

### CODE REVIEW

**Comment by anjor** [comment]
@aglick24 thanks for the contribution!

It looks like there is some dispute about the precise format though -- https://huggingface.co/junelee/wizard-vicuna-13b/discussions/1#6453c515b8c58783d662bc05 


Also I think you will need to import it in the __init__.py like so: https://huggingface.co/junelee/wizard-vicuna-13b/discussions/1#6453c515b8c58783d662bc05

**Comment by aglick24** [comment]
Thanks @anjor ! What do you think about this format? On lines 393-404 https://github.com/lm-sys/FastChat/blob/main/fastchat/conversation.py and if so, should I include the </s> separator?

**Comment by aglick24** [comment]
This seems to me to be an example of what that conversation template should look like:

"A chat between a curious user and an artificial intelligence assistant. The assistant gives helpful, detailed, and polite answers to the user's questions. USER: What is ChatGPT? ASSISTANT: ChatGPT is an advanced language model developed by OpenAI. It's designed to understand and generate natural language, so it can converse with users, answer questions, and even assist with various tasks.</s>USER: So, it's like a super-advanced chatbot? ASSISTANT: Yes, but it can engage in broader conversations and more complex reasoning.</s>"

**Comment by aglick24** [comment]
Following up @anjor

**Comment by anjor** [comment]
Could you fix the tests?

Also, if you could hold off until https://github.com/deployradiant/pychatml/pull/14 is merged. It might be a good idea to implement that abstract class instead of the current implementation. There is an example implementation in https://github.com/deployradiant/pychatml/pull/15

File: pychatml/vicuna_v1_1/vicuna_v1_1.py
Comment (anjor) [inline pychatml/vicuna_v1_1/vicuna_v1_1.py:90]: let's remove this. The test is good enough.

File: pychatml/vicuna_v1_1/vicuna_v1_1.py
Comment (anjor) [inline pychatml/vicuna_v1_1/vicuna_v1_1.py:24]: why is this needed?

Comment (anjor) [inline pychatml/vicuna_v1_1/vicuna_v1_1.py:24]: is this just to convert to a list?

Comment (aglick24) [reply pychatml/vicuna_v1_1/vicuna_v1_1.py:24]: yes! That way I can index into it

File: pychatml/vicuna_v1_1/vicuna_v1_1.py
Comment (aglick24) [reply pychatml/vicuna_v1_1/vicuna_v1_1.py:90]: yep, sorry forgot about it!

**Comment by koogle** [comment]
@aglick24  let us know if there is anything we can help with to get it over the line and merged?

**Comment by aglick24** [comment]
Thanks @koogle for merging main! Sorry about the delay, I assumed we were waiting on the changes to merge to main before I would keep going. Looking through it now to update it to the new format

**Comment by aglick24** [comment]
Just fixed it all and the tests! Created version 0.0.6 to add vicuna

File: tests/test_llama2.py
Comment (koogle) [inline tests/test_llama2.py:11]: is that change intentional?

**Comment by koogle** [comment]
@aglick24 thank you so much for this contribution, approving

**Review by koogle — APPROVED** [review]

**Comment by aglick24** [comment]
Screenshot 2024-01-03 at 9 40 06 AM

Thanks so much for the approval @koogle ! Unfortunately, I need write access to merge

**Comment by koogle** [comment]
Okay in that case I will merge it , thank you for the contribution

**Comment by koogle** [comment]
@aglick24  released 0.0.7 https://github.com/deployradiant/pychatml/releases/tag/0.0.7

**Pull request merged**
